homeassistant/components/history_stats/data.py

```diff
@@ -110,6 +110,21 @@ async def async_update(
                 )
             )
         ):
+            start_changed = (
+                current_period_start_timestamp != previous_period_start_timestamp
+            )
+            if start_changed:
+                # Trim the start of the _history_current_period to the new period start
+                trim_count = 0
+                for i, state in enumerate(self._history_current_period):
+                    if state.last_changed < current_period_start_timestamp:
+                        state.last_changed = current_period_start_timestamp
```

Comment:
I'd make this its own function with a doc string that explains whats going on here so we can better show intent to future maintainers